### PR TITLE
Added Wind Module

### PR DESCRIPTION
**Pull request by sericson0**
Converted wind code from REopt API to Julia. I use PyCall to interface with the sscapi to download wind production factors

### CODE REVIEW

**Comment by NLaws** [comment]
@sericson0  looks great! Can you please run `add PyCall`, as well add any other dependencies that are needed for these changes? (I can see that PyCall is why the [tests are currently failing](https://github.com/NREL/REoptLite/runs/2781651705?check_suite_focus=true#step:5:146)).  [Here](https://pkgdocs.julialang.org/v1/creating-packages/#Adding-dependencies-to-the-project) is the documentation on adding package dependencies.

File: src/Manifest.toml [hunk 1]
```diff
@@ -0,0 +1,203 @@
+# This file is machine-generated - editing it directly is not advised
```
Comment (NLaws) [inline src/Manifest.toml:1]: Is this Manifest.toml necessary?

File: src/Project.toml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[deps]
```
Comment (NLaws) [inline src/Project.toml:1]: If the src/Manifest.toml is not necessary then neither is this Project.toml

File: src/REoptLite.jl [hunk 3]
```diff
@@ -41,7 +41,10 @@ export
     Financial,
     add_financial_results,
     ElectricUtility,
-    Generator
+    Generator,
+    #Added Wind
```
Comment (NLaws) [inline src/REoptLite.jl:45]: Please remove the "Added Wind" comments throughout

**Review by NLaws — CHANGES_REQUESTED** [review]
I put some inline comments. There are a lot of "Added Wind" comments that should be removed. Also please remove the .pyc files. I think that in general it all looks good!

**Comment by NLaws** [comment]
@sericson0 
When I run the new wind test I get:
```julia
ArgumentError: ref of NULL PyObject
  Stacktrace:
    [1] _getproperty
      @ ~/.julia/packages/PyCall/BD546/src/PyCall.jl:299 [inlined]
    [2] __getproperty(o::PyCall.PyObject, s::Symbol)
      @ PyCall ~/.julia/packages/PyCall/BD546/src/PyCall.jl:306
    [3] getproperty
      @ ~/.julia/packages/PyCall/BD546/src/PyCall.jl:312 [inlined]
    [4] make_ssc(use_input_data::Bool, file_resource_full::String, d::Dict{String, Any})
      @ REoptLite ~/projects/reopt/seansfork/src/core/wind_prodfactor.jl:178
    [5] sam_wind_prod_factors(wind::Wind, time_steps_per_hour::Float64, latitude::Float64, longitude::Float64)
      @ REoptLite ~/projects/reopt/seansfork/src/core/wind_prodfactor.jl:6
    [6] setup_wind_inputs(s::Scenario, time_steps::UnitRange{Int64}, max_sizes::JuMP.Containers.DenseAxisArray{Float64, 1, Tuple{Vector{String}}, Tuple{JuMP.Containers._AxisLookup{Dict{String, Int64}}}}, min_sizes::JuMP.Containers.DenseAxisArray{Float64, 1, Tuple{Vector{String}}, Tuple{JuMP.Containers._AxisLookup{Dict{String, Int64}}}}, existing_sizes::JuMP.Containers.DenseAxisArray{Float64, 1, Tuple{Vector{String}}, Tuple{JuMP.Containers._AxisLookup{Dict{String, Int64}}}}, cap_cost_slope::JuMP.Containers.DenseAxisArray{Float64, 1, Tuple{Vector{String}}, Tuple{JuMP.Containers._AxisLookup{Dict{String, Int64}}}}, om_cost_per_kw::JuMP.Containers.DenseAxisArray{Float64, 1, Tuple{Vector{String}}, Tuple{JuMP.Containers._AxisLookup{Dict{String, Int64}}}}, production_factor::JuMP.Containers.DenseAxisArray{Float64, 2, Tuple{Vector{String}, UnitRange{Int64}}, Tuple{JuMP.Containers._AxisLookup{Dict{String, Int64}}, JuMP.Containers._AxisLookup{Tuple{Int64, Int64}}}})
      @ REoptLite ~/projects/reopt/seansfork/src/core/reopt_inputs.jl:320
    [7] setup_tech_inputs(s::Scenario)
      @ REoptLite ~/projects/reopt/seansfork/src/core/reopt_inputs.jl:209
    [8] REoptInputs(s::Scenario)
      @ REoptLite ~/projects/reopt/seansfork/src/core/reopt_inputs.jl:98
 ```
 Note that I moved the wind test into test_with_cbc.jl, which is what is run on Github Actions here and you can run by starting julia from your repo's directory with `julia --project=.`:
```
[~/projects/reopt/seansfork]
nlaws-> julia --project=.
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.6.1 (2021-04-23)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

julia> ]

(REoptLite) pkg> test
     Testing REoptLite
...
```

**Comment by NLaws** [comment]
Maybe it would be better to wrap the precompiled SAM C++ code with:
https://github.com/JuliaInterop/CxxWrap.jl 
?

**Comment by sericson0** [comment]
That  could  definitely  be  good. All the PyCall  stuff was just  to be
able  to  connect  with  SAM and would be  much  cleaner to  have it all in
Julia. I can take  a look  next week  if that  is ok

**Comment by NLaws** [comment]
Figured out how to use `ccall` with SAM libraries to remove the reliance on `PyCall` in https://github.com/NREL/REoptLite/pull/45

**Pull request closed without merging**